api/src/wfl/service/rawls.clj

```diff
@@ -37,14 +39,62 @@
    (create-snapshot-reference workspace snapshot-id name "")))
 
 (defn get-snapshot-reference
-  "Return the snapshot reference in fully-qualified Terra WORKSPACE with REFERENCE-ID."
+  "Return the snapshot reference in `workspace` with `reference-id`."
   [workspace reference-id]
-  (get-workspace-json workspace "snapshots" reference-id))
+  (get-workspace-json workspace snapshot-endpoint reference-id))
+
+(defn get-snapshot-references
+  "Lazily returns the snapshot references in `workspace`
+  with `limit` resources per page (default: 100)."
+  ([workspace limit]
+   (letfn
+    [(page [offset]
+       (let [{:keys [resources] :as _response}
+             (-> (workspace-api-url workspace snapshot-endpoint)
+                 (http/get {:headers      (auth/get-auth-header)
+                            :query-params {:offset offset
+                                           :limit  limit}})
+                 (util/response-body-json))
+             total    (+ offset (count resources))]
+         (lazy-cat resources (when (seq resources) (page total)))))]
+     (util/lazy-unchunk (page 0))))
+  ([workspace]
+   (get-snapshot-references workspace 100)))
```

Comment:
Do we need `lazy-unchunk`?
Cannot hurt much, I guess.

Reply:
I think so, our current use case for this method is to find the first reference matching a snapshot ID.  Not very many snapshot references to look through now, but if we retain them indefinitely as is our current plan we could experience some savings.  (Bigger than this use case, I was hoping down the road we might generalize this to other cases where lazy pagination could be an asset, like [`datarepo/list-snapshots`](https://github.com/broadinstitute/wfl/blob/develop/api/src/wfl/service/datarepo.clj#L163-L182).)